xarray/core/groupby.py

```diff
@@ -362,6 +362,22 @@ def _maybe_unstack(self, obj):
                     del obj.coords[dim]
         return obj
 
+    def _maybe_stack(self, applied):
```

Comment:
This method becomes necessary, because now we cannot do `xr.concat([ds.isel(yx=i)] for i in range(*)], dim='yx')` because `ds.isel(yx=i)` does not have `yx` anymore.